### ISSUE TITLE
onToggle is not invoked when pressing enter key on Toggle

### ISSUE THREAD

**Issue by lisadier** [carbon-design-system/carbon#3453]
## Environment

> Operating system
Mac and others

> Browser
Chrome and others

> Automated testing tool and ruleset
n/a

> Assistive technology used to verify
just normal keyboard

## Detailed description
When the user toggles the Toggle component the view updates and the onToggle handler is called if the user clicks the toggle or uses the space bar for toggling.  However, using the enter key to toggle, results in the view being update but no call to the onToggle handler.  I noticed that there is an exception in the Chrome DevTools console:
Uncaught TypeError: _onChange is not a function
    at onKeyUp (Toggle.js:108)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
    at invokeGuardedCallback (react-dom.development.js:256)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:270)
    at executeDispatch (react-dom.development.js:561)
    at executeDispatchesInOrder (react-dom.development.js:583)
    at executeDispatchesAndRelease (react-dom.development.js:680)
    at executeDispatchesAndReleaseTopLevel (react-dom.development.js:688)
    at forEachAccumulated (react-dom.development.js:662)
    at runEventsInBatch (react-dom.development.js:816)
    at runExtractedEventsInBatch (react-dom.development.js:824)
    at handleTopLevel (react-dom.development.js:4826)
    at batchedUpdates$1 (react-dom.development.js:20439)
    at batchedUpdates (react-dom.development.js:2151)
    at dispatchEvent (react-dom.development.js:4905)
    at eval (react-dom.development.js:20490)
    at Object.unstable_runWithPriority (scheduler.development.js:255)
    at interactiveUpdates$1 (react-dom.development.js:20489)
    at interactiveUpdates (react-dom.development.js:2170)
    at dispatchInteractiveEvent (react-dom.development.js:4882)

> What version of the Carbon Design System are you using?
"carbon-components": "^10.3.1",
"carbon-components-react": "^7.3.2",

> What did you expect to happen?
onToggle should be called when pressing enter key, same as clicking and pressing space bar

> What happened instead?
onToggle was not called when pressing enter key

> What WCAG 2.1 checkpoint does the issue violate?
keyboard accessibility

## Steps to reproduce the issue
1. Press enter on the toggle
2. Notice that the onToggle was not invoked and an exception is logged in the console.

> Please create a reduced test case in CodeSandbox
https://codesandbox.io/s/codesandbox-2vk1e

## Additional information
If you bring up the Chrome devTools console, there is an exception logged when clicking enter that doesn't happen when clicking or pressing space bar.
![image](https://user-images.githubusercontent.com/16292545/61484502-afa28c00-a964-11e9-94ec-404318ea7b0c.png)
